watertap/examples/flowsheets/RO_with_energy_recovery/RO_with_energy_recovery.py

```diff
@@ -307,65 +350,78 @@ def initialize_system(m, solver=None):
     # ---initialize feed block---
     m.fs.feed.initialize(optarg=optarg)
 
-    # ---initialize splitter and pressure exchanger---
-    # pressure exchanger high pressure inlet
-    propagate_state(m.fs.s06)  # propagate to PXR high pressure inlet from RO retentate
-    m.fs.PXR.high_pressure_side.properties_in.initialize(optarg=optarg)
-
-    # splitter inlet
-    propagate_state(m.fs.s01)  # propagate to splitter inlet from feed
-    m.fs.S1.mixed_state.initialize(
-        optarg=optarg
-    )  # initialize inlet state block to solve for mass fraction
-    # splitter outlet to PXR, enforce same volumetric flow as PXR high pressure inlet
-    m.fs.S1.PXR_state.calculate_state(
-        var_args={
-            (
-                "flow_vol_phase",
-                "Liq",
-            ): value(  # same volumetric flow rate as PXR high pressure inlet
-                m.fs.PXR.high_pressure_side.properties_in[0].flow_vol_phase["Liq"]
-            ),
-            ("mass_frac_phase_comp", ("Liq", "NaCl")): value(
-                m.fs.S1.mixed_state[0].mass_frac_phase_comp["Liq", "NaCl"]
-            ),  # same as splitter inlet
-            ("pressure", None): value(
-                m.fs.S1.mixed_state[0].pressure
-            ),  # same as splitter inlet
-            ("temperature", None): value(m.fs.S1.mixed_state[0].temperature),
-        },  # same as splitter inlet
-    )
-    # splitter initialization
-    m.fs.S1.PXR_state[0].flow_mass_phase_comp[
-        "Liq", "NaCl"
-    ].fix()  # fix the single degree of freedom for unit
-    m.fs.S1.initialize(optarg=optarg)
-    m.fs.S1.PXR_state[0].flow_mass_phase_comp[
-        "Liq", "NaCl"
-    ].unfix()  # unfix for flowsheet simulation and optimization
-
-    # pressure exchanger low pressure inlet
-    propagate_state(m.fs.s08)
-
-    # pressure exchanger initialization
-    m.fs.PXR.initialize(optarg=optarg)
-
-    # ---initialize pump 1---
-    propagate_state(m.fs.s02)
-    m.fs.P1.initialize(optarg=optarg)
-
-    # ---initialize pump 2---
-    propagate_state(m.fs.s09)
-    m.fs.P2.control_volume.properties_out[0].pressure.fix(
-        value(m.fs.P2.control_volume.properties_out[0].pressure)
-    )
-    m.fs.P2.initialize(optarg=optarg)
-    m.fs.P2.control_volume.properties_out[0].pressure.unfix()
+    if m.erd_type == "pressure_exchanger":
+        # ---initialize splitter and pressure exchanger---
```

Comment:
Should we just write a separate initialize function for each configuration, and call the appropriate function?

Reply:
This has been addressed by reorganizing the flowsheet to create/initialize the main pump, RO, and then an option for the ERD configuration. For initialization, the option now calls `initialize_pressure_exchanger` or `initialize_pump_as_turbine`.